### PR TITLE
[cicd] add phase 2 proposal for the CICD SIG

### PR DESCRIPTION
**Pull request by adrielp**

### CODE REVIEW

File: projects/ci-cd-phase-2.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+- [ ] Platform integration testing resources
+
+**Success Dependencies**:
+- Dedicated engineering time from SDK teams (guesstimated 2-3 months per SDK)
+- Platform vendor collaboration for integration testing
+- Community feedback and early adoption validation
+
+## Meeting Times
+
+This SIG will meeting weekly on Wednesdays at 06:00 PT. (Tentative)
```
Comment (horovits) [inline projects/ci-cd-phase-2.md]: pending community pole on preferred times (that is planned to after folks come back from summer vacations)

File: projects/ci-cd-phase-2.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+This phase targets three core areas for immediate impact:
+
+1. **SDK Environment Variable Context Propagation**: Implement the spec across
+   multiple OpenTelemetry languages for first class, native support.
+2. **Enhanced Signal Attribute Conventions**: Establish additional attribute
+   conventions for CI/CD-specific signals including build metadata, deployment
+   context, issues, incidents, and pipeline execution details.
+3. **Beta Stabilization**: Move core CI/CD semantic conventions and SDK
+   implementations from experimental to beta status
+4. **Increase Adoption**: Work towards getting one or more key technologies
```
Comment (horovits) [inline projects/ci-cd-phase-2.md]: I'd rephrase this into two separate goals:
4. New Adoption: Work towards getting... (same phrasing). - Do we want to name specific technologies/tools/projects?
5. Expand adoption within the OpenTelemetry project: building on the success of phase 1 with a pilot adoption of the SemConv for contrib which proved useful on multiple investigations by SIG Infra, expand the adoption to more parts of the project, "dog fooding", providing further feedback to the SIG, and then providing reference to other projects (per goal 4 above)

File: projects/ci-cd-phase-2.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+### Phase Two Goals
+
+This phase targets three core areas for immediate impact:
+
+1. **SDK Environment Variable Context Propagation**: Implement the spec across
+   multiple OpenTelemetry languages for first class, native support.
+2. **Enhanced Signal Attribute Conventions**: Establish additional attribute
+   conventions for CI/CD-specific signals including build metadata, deployment
+   context, issues, incidents, and pipeline execution details.
```
Comment (kamphaus) [inline projects/ci-cd-phase-2.md]: Currently we have not yet defined SemConv for logs or events.
These should definitely be part of phase 2.

Here we could also add conventions for test results.

File: projects/ci-cd-phase-2.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+3. **Beta Stabilization**: Move core CI/CD semantic conventions and SDK
+   implementations from experimental to beta status
```
Comment (kamphaus) [inline projects/ci-cd-phase-2.md]: There is still the open issue around long running traces: https://github.com/open-telemetry/semantic-conventions/issues/1648
Do you think that could be a blocker for stabilization?
Should it be addressed in phase 2?

File: projects/ci-cd-phase-2.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+4. **Increase Adoption**: Work towards getting one or more key technologies
+   adoption of the OpenTelemetry conventions.
+
+## Phase 2 Staffing & Responsibilities
```
Comment (danielgblanco) [inline projects/ci-cd-phase-2.md:56]: Sign me up as GC liaison, as I'm already the one for this SIG.

File: projects/ci-cd-phase-2.md (hunk 2 above)
Comment (danielgblanco) [reply projects/ci-cd-phase-2.md]: I agree, I think we should aim to be more specific here.

> Do we want to name specific technologies/tools/projects?

It may be worth reaching out as part of this project proposal. I can help, if we have some technologies/tools in mind.

> Expand adoption within the OpenTelemetry project

Same here. I love the idea, but maybe we can list where we'd like to adopt it, so that folks sign up to it.

File: projects/ci-cd-phase-2.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+There is still a lot of things to accomplish, and as such, we propose
+implementing phase two of the CI/CD Observability SIG.
+
+### Phase Two Goals
+
+This phase targets three core areas for immediate impact:
+
+1. **SDK Environment Variable Context Propagation**: Implement the spec across
+   multiple OpenTelemetry languages for first class, native support.
```
Comment (danielgblanco) [inline projects/ci-cd-phase-2.md:34]: Do we want to list the languages we want to target? Will this be contributions from this SIG to those other languages? Or would you expect other SIGs/contributors to join the effort and contribute?

File: projects/ci-cd-phase-2.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+**Project Leadership**:
+* @horovits Project Lead (AWS)
+* @adrielp Project Lead (Liatrio)
+
+### Phase 2 Resource Requirements
+
+**Critical Needs**:
+- [ ] SDK maintainer commitment for environment variable context propagation
+  implementation
```
Comment (danielgblanco) [inline projects/ci-cd-phase-2.md:84]: Good call. If we have those for a collection of languages, we can list them above.

File: projects/ci-cd-phase-2.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+## Phase 2 Staffing & Responsibilities
+
+### TC Sponsorship
+
+<!-- TODO: Vet that they still want to be sponsors -->
+* @carlosalberto (TC Sponsor)
+* @jsuereth (TC Sponsor)
+
+### Phase 2 Team Structure
```
Comment (danielgblanco) [inline projects/ci-cd-phase-2.md:66]: Can we add a subsection here for other contributors that want to join this effort?

cc @martincostello

File: projects/ci-cd-phase-2.md (hunk 2 above)
Comment (adrielp) [reply projects/ci-cd-phase-2.md]: I'm thinking that for new technologies, we focus on adoption of the Env context in the OTEL APIs for Python, Go, and one other language (maybe c# since runners in GitHub are written in c#?).  And maybe we work on getting Argo Workflows Adoption?

For expanded adoption, I'm not sure. It'd be good to know where Team City & Jenkins sits and whether or not they are/feel done.

File: projects/ci-cd-phase-2.md (hunk 4 above)
Comment (adrielp) [reply projects/ci-cd-phase-2.md]: I don't think so -- but happy to get input from others.

File: projects/ci-cd-phase-2.md (hunk 2 above)
Comment (martincostello) [reply projects/ci-cd-phase-2.md]: > maybe c# since runners in GitHub are written in c#

I'm happy to get involved with that as that's my primary language

File: projects/ci-cd-phase-2.md (hunk 6 above)
Comment (adrielp) [reply projects/ci-cd-phase-2.md:34]: I'm adding languages now and am wondering if we should do all the major languages.

File: projects/ci-cd-phase-2.md (hunk 2 above)
Comment (horovits) [reply projects/ci-cd-phase-2.md]: that'd be splendid @martincostello !

File: projects/ci-cd-phase-2.md (hunk 4 above)
Comment (horovits) [reply projects/ci-cd-phase-2.md]: I also don't think so. if @danielgblanco / oversight sees it as blocker to wrap up phase 1, we can push to phase 2 and carry on there IMO

File: projects/ci-cd-phase-2.md [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+
+This phase targets three core areas for immediate impact:
+
+1. **SDK Environment Variable Context Propagation**: Implement the spec across
+   multiple OpenTelemetry languages for first class, native support.
+   - Python
+   - Go
+   - C#
+   - Javascript
+2. **Enhanced Signal Attribute Conventions**: Establish additional attribute
```
Comment (carlosalberto) [inline projects/ci-cd-phase-2.md:39]: Some of these items are very clear (logs/events, test results) but some other could get additional details/examples.

File: projects/ci-cd-phase-2.md (hunk 4 above)
Comment (carlosalberto) [reply projects/ci-cd-phase-2.md]: IMHO this is a potential thing the SIG could help with, depending on how the discussion moves - but in general this "limitation" has been a well known issue for some time now.

My question would be: how often this affects **existing** components? Collector components, Jenkins plugins, etc? Or is this more of a (relatively) corner case?

File: projects/ci-cd-phase-2.md [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+   conventions for CI/CD-specific signals including build metadata, deployment
+   context, issues, incidents, logs/events, test results, and pipeline execution
+   details.
+3. **Beta Stabilization**: Move core CI/CD semantic conventions and SDK
```
Comment (carlosalberto) [inline projects/ci-cd-phase-2.md]: What SDKs are you referring to here?

File: projects/ci-cd-phase-2.md [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+   implementations from experimental to beta status
+4. **Increase Adoption**: Work towards getting one or more new technologies
+   adoption of the OpenTelemetry conventions.
+5. **Extend Adoption**:
```
Comment (carlosalberto) [inline projects/ci-cd-phase-2.md]: Add details? Otherwise it can be left out (IMHO there's enough work listed already, etc)

**Comment by svrnm** [comment]
@adrielp can you take a look into the outstanding conversations?

**Comment by adrielp** [comment]
> @adrielp can you take a look into the outstanding conversations?

Yes, absolutely. I've been out of band during a transition and am starting to be able to get bandwidth again. Will take a look and push any necessary updates.

File: projects/ci-cd-phase-2.md (hunk 1 above)
Comment (adrielp) [reply projects/ci-cd-phase-2.md]: New poll is up https://strawpoll.com/2ayLQBE1en4

File: projects/ci-cd-phase-2.md (hunk 11 above)
Comment (adrielp) [reply projects/ci-cd-phase-2.md]: Opting to leave out for phase 2.

File: projects/ci-cd-phase-2.md (hunk 10 above)
Comment (adrielp) [reply projects/ci-cd-phase-2.md]: removed

File: projects/ci-cd-phase-2.md (hunk 9 above)
Comment (adrielp) [reply projects/ci-cd-phase-2.md:39]: I've adjusted the phrasing to make it more clear. Let me know if you think there are additional changes needed.

File: projects/ci-cd-phase-2.md (hunk 4 above)
Comment (adrielp) [reply projects/ci-cd-phase-2.md]: I've added a stretch goal to help bring this forward. Let me know if anything should adjusted.

**Comment by adrielp** [comment]
I've pushed changes and addressed comments. A new poll is out to find the time for the meeting. Set the first week of November. https://strawpoll.com/2ayLQBE1en4

File: projects/ci-cd-phase-2.md [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+
+### GC Liaison
+
+* @danielgblanco (GC Liaison)
+
+### TC Sponsorship
+
+<!-- TODO: Vet that they still want to be sponsors -->
+* @carlosalberto (TC Sponsor)
+* @jsuereth (TC Sponsor)
```
Comment (trask) [inline projects/ci-cd-phase-2.md]: cc @carlosalberto @jsuereth

File: projects/ci-cd-phase-2.md [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+- [ ] Platform integration testing resources
+
+**Success Dependencies**:
+- Dedicated engineering time from SDK teams (guesstimated 2-3 months per SDK)
+- Platform vendor collaboration for integration testing
+- Community feedback and early adoption validation
+
+## Meeting Times
+
+This SIG will meeting weekly on Wednesdays at 06:00 PT. (Tentative)
```
Comment (kamphaus) [inline projects/ci-cd-phase-2.md]:
```suggestion
This SIG will meet weekly on Wednesdays at 06:00 PT. (Tentative)
```

File: projects/ci-cd-phase-2.md [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+
+## Phase 2 Timeline
+
+**Duration**: 6 months (targeting completion by end of 2025)
```
Comment (kamphaus) [inline projects/ci-cd-phase-2.md]: 3 months remaining in 2025. To be adapted?

File: projects/ci-cd-phase-2.md [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+
+> Note: Timeline is aggressive but achievable with focused scope and dedicated
```
Comment (kamphaus) [inline projects/ci-cd-phase-2.md]:
```suggestion
> Note: The timeline is aggressive but achievable with focused scope and dedicated
```

File: projects/ci-cd-phase-2.md [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+This phase targets three core areas for immediate impact:
+
+1. **SDK Environment Variable Context Propagation**: Implement the spec across
+   multiple OpenTelemetry languages for first class, native support.
+   - Python
+   - Go
+   - C#
+   - Javascript
+2. **Enhanced Signal Attribute Conventions**: Establish additional attribute
+   conventions for CI/CD-specific trace, metrics, and logs for the following
```
Comment (kamphaus) [inline projects/ci-cd-phase-2.md]: I would list events instead of logs since we cannot make any more improvements for logs beyond https://github.com/open-telemetry/semantic-conventions/pull/2752.

With events we should strive towards compatibility/interoperability with [CDEvents](https://cdevents.dev/).
Events could also be part of a solution for the long running trace issue.

```suggestion
   conventions for CI/CD-specific trace, metrics, and events for the following
```

Comment (adrielp) [reply projects/ci-cd-phase-2.md]: A log = event. Log is the signal, unless that definition has changed.

> With events we should strive towards compatibility/interoperability with [CDEvents](https://cdevents.dev/).

I'm not sure we should, but we should definitely actively consider & evaluate. When we made the initial proposal it seemed like there wasn't much appetite there. Would love the folks over there to contribute though and figure out where we can be interoperable.

Comment (kamphaus) [reply projects/ci-cd-phase-2.md]: While in the protocol log = event, as I understood it the difference is in SemConv:
A log can be structured or unstructured but it does not have a type/name. Thus we cannot define SemConv for it.
Events do have a name and we can define SemConv for events with a specific name.

Comment (horovits) [reply projects/ci-cd-phase-2.md]: > I would list events instead of logs

@adrielp @kamphaus both of you are correct.
Log is the telemetry signal type. 
Event is a log record with a mandatory name and a structure that is defined by a schema.
https://opentelemetry.io/docs/concepts/glossary/#event
At OTel we advocate for log records to be structured as events.
Let's disambiguate the above, I suggest:
`conventions for CI/CD-specific the trace, metrics, and logs telemetry signal types for the following`

> With events we should strive towards compatibility/interoperability with [CDEvents](https://cdevents.dev/).

I agree with @adrielp , the discussion so far was definitely not conclusive on either CDEvents side nor OTel side to place this is as a goal.

File: projects/ci-cd-phase-2.md [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+
+* @danielgblanco (GC Liaison)
+
+### TC Sponsorship
+
+<!-- TODO: Vet that they still want to be sponsors -->
+* @carlosalberto (TC Sponsor)
+* @jsuereth (TC Sponsor)
+
+### Phase 2 Team Structure
```
Comment (horovits) [inline projects/ci-cd-phase-2.md:66]: should we also list approvers? @danielgblanco

File: projects/ci-cd-phase-2.md [hunk 18]
```diff
@@ -0,0 +1,104 @@
...
+### Phase 2 Resource Requirements
+
+**Critical Needs**:
+- [ ] SDK maintainer commitment for environment variable context propagation
+  implementation
+- [ ] Platform integration testing resources
+
+**Success Dependencies**:
+- Dedicated engineering time from SDK teams (guesstimated 2-3 months per SDK)
+- Platform vendor collaboration for integration testing
```
Comment (carlosalberto) [inline projects/ci-cd-phase-2.md:89]: Anybody in mind? Curious about this but not a blocker.

File: projects/ci-cd-phase-2.md (hunk 12 above)
Comment (svrnm) [reply projects/ci-cd-phase-2.md]: @jsuereth please take a look and confirm if you want to be a sponsor

File: projects/ci-cd-phase-2.md (hunk 17 above)
Comment (adrielp) [reply projects/ci-cd-phase-2.md:66]: Added codeowners group as approvers

File: projects/ci-cd-phase-2.md (hunk 18 above)
Comment (adrielp) [reply projects/ci-cd-phase-2.md:89]: In mind, yes, folks I want to include here, not yet. We have some comms/bridges to build.

File: projects/ci-cd-phase-2.md (hunk 12 above)
Comment (adrielp) [reply projects/ci-cd-phase-2.md]: spoke with both, Carlos will be TC sponsor. AFAIK we should only need one. I have removed Josh.

**Review by jsuereth — APPROVED** [review]
Just wanted to comment that I'm glad to see this work continue!

If Carlos is able to be the full TC sponsor, I'm happy to continue supporting via OTEL Semantic Convention discussions and code reviews.  Thanks for continuing this work!

**Review by trask — APPROVED** [review]

**Review by mtwo — APPROVED** [review]

**Review by jpkrohling — APPROVED** [review]

File: projects/ci-cd-phase-2.md [hunk 19]
```diff
@@ -0,0 +1,105 @@
+# CI/CD Observability SIG Phase 2
+
+<!-- TODO: Potentially create a new project board -->
```
Comment (danielgblanco) [inline projects/ci-cd-phase-2.md:3]: I think that'd be good, as it'd align better with the Roadmap Management process. Each "board" is a different project, with different target dates and so on.

File: projects/ci-cd-phase-2.md [hunk 20]
```diff
@@ -0,0 +1,105 @@
...
+* @horovits Project Lead (AWS)
+* @adrielp Project Lead (Liatrio)
+
+**Approvers**:
+* @open-telemetry/semconv-cicd-approvers
+
+**Project Members**:
+* @martincostello
+* @kamphaus
+* @niwoerner
```
Comment (niwoerner) [inline projects/ci-cd-phase-2.md]:
```suggestion
* @niwoerner (OllyGarden)
```

File: projects/ci-cd-phase-2.md [hunk 21]
```diff
@@ -0,0 +1,105 @@
...
+
+**Project Leadership**:
+* @horovits Project Lead (AWS)
+* @adrielp Project Lead (Liatrio)
+
+**Approvers**:
+* @open-telemetry/semconv-cicd-approvers
+
+**Project Members**:
+* @martincostello
```
Comment (martincostello) [inline projects/ci-cd-phase-2.md]:
```suggestion
* @martincostello (Grafana Labs)
```

**Comment by trask** [comment]
@adrielp all good to merge? I can hit the button, thanks

File: projects/ci-cd-phase-2.md (hunk 19 above)
Comment (danielgblanco) [reply projects/ci-cd-phase-2.md:3]: Let's do this after the project starts
